### ISSUE TITLE
Running Parallel Programs

### ISSUE THREAD

**Issue by Singh-Krishna7** [microsoft/vscode#146508]
Issue Type: <b>Performance Issue</b>

Below is my current version information after which i started observing the issue:
Version: 1.66.0 (user setup)
Commit: e18005f0f1b33c29e81d732535d8c0e47cafb0b5
Date: 2022-03-30T05:50:14.623Z
Electron: 17.2.0
Chromium: 98.0.4758.109
Node.js: 16.13.0
V8: 9.8.177.11-electron.0
OS: Windows_NT x64 10.0.22000

Earlier could run multiple instances of Visual Studio Code, however post last update not able to run instances smoothly.

VS Code version: Code 1.66.0 (e18005f0f1b33c29e81d732535d8c0e47cafb0b5, 2022-03-30T05:50:14.623Z)
OS version: Windows_NT x64 10.0.22000
Restricted Mode: No

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-10510U CPU @ 1.80GHz (8 x 2304)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: disabled_off<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>oop_rasterization: enabled<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>skia_renderer: enabled_on<br>video_decode: enabled<br>video_encode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|undefined|
|Memory (System)|15.77GB (3.67GB free)|
|Process Argv|<Hiding this directory info for privacy> --crash-reporter-id 11ab182f-fe61-41f1-b61b-6925d8255792|
|Screen Reader|no|
|VM|0%|
</details><details><summary>Extensions (4)</summary>

Extension|Author (truncated)|Version
---|---|---
csharp|ms-|1.24.1
azure-account|ms-|0.10.1
powershell|ms-|2021.12.0
usql-vscode-ext|usq|0.2.15

</details><details>
<summary>A/B Experiments</summary>

```
vsliv368cf:30146710
vsreu685:30147344
python383:30185418
vspor879:30202332
vspor708:30202333
vspor363:30204092
pythonvspyl392:30443607
pythontb:30283811
pythonptprofiler:30281270
vshan820:30294714
vstes263:30335439
vscorecescf:30445987
pythondataviewer:30285071
vscod805cf:30301675
pythonvspyt200:30340761
binariesv615:30325510
bridge0708:30335490
bridge0723:30353136
vsaa593cf:30376535
vsc1dst:30438360
pythonvs932:30410667
wslgetstarted:30449410
vsclayoutctrt:30451275
pythonvsnew555:30457759
vscscmwlcmc:30438804
pynewfile477:30463512

```

</details>

**Comment by IllusionMH** [comment]
> however post last update not able to run instances smoothly.

Please elaborate what is the problem with multiple instances

**Comment by Singh-Krishna7** [comment]
_**Prior to Update(1.66.0):**_ 
      -  Opening two instances of same repository at two different locations would happen with ease.(For Ex: Having a clone 
                  of a repo A in two locations in Windows(running windows 11) machine was no issues; Loc1= C:\MyProj\A and Loc2= 
                C:\MyProj2\A.I could open two VSCode instance and alter or edit codes in them parallelly.
_**Post to Update(1.66.0):**_  
      -  Able to start two VSCode instances parallely, however the lag is Huge, Tried checking the Processes, MemoryUsage, CPU 
                 usage and Disc usage, No sudden change in them.  When i say lag is huge, its like the VSCode instances are 
                 irresponsive until unless 5seconds of click.
      
_**New Observation:**_
      - After leaving the terminals open for around 20-30mins and not performing any operations, stabilizes this lag. 
      
Observed the lag issue on multiple occasions and hence created this ticket after repro'ing the issue multiple times.

**Comment by bpasero** [comment]
Can you follow this wiki and post the results back here to understand more about why performance is so bad for you: https://github.com/Microsoft/vscode/wiki/Performance-Issues

Thanks!

**Comment by Singh-Krishna7** [comment]
Also, I've been observing weird issue while trying to resolve the git conflicts in studio code. Not sure if the behavior for the conflict resolution is changed or its a bug.  Screen shot for reference:
<img width="370" alt="image" src="https://user-images.githubusercontent.com/77609221/161943061-4af64063-325a-49ec-8087-4fd598e68243.png">
Also, the actual conflict is below: 
<img width="412" alt="image" src="https://user-images.githubusercontent.com/77609221/161943868-85cf7afc-9969-4ca1-b818-2da2361e8376.png">

**Comment by lorand-horvath** [comment]
@Singh-Krishna7 Can we please try to solve the lag issues in this ticket? You should open a different issues regarding the git conflicts.
Please also see this issue regarding a possible solution to the lag/slowness: https://github.com/microsoft/vscode/issues/146737#issuecomment-1090034221

**Comment by bpasero** [comment]
Most likely a duplicate of https://github.com/microsoft/vscode/issues/146737

**Comment by Singh-Krishna7** [comment]
> 

@lorand-horvath sure, can create a separate ticket for that, however was trying to bring to focus as to what issues were faced along with slow tab switching for editing.

**Comment by Singh-Krishna7** [comment]
> /gifPlease

I cannot show the code section as it would cause a compliance issue neither do i have any demo code to show it via gif, none the less here is what i do to reach that state:
1) When trying to merge user branch to master/main branch, if there are conflicts via the CMD prompt, i used VScode to resolve conflicts. However, when i open the vscode to resolve the conflict from the CMD terminal with "code ." command, the above options show in VSCode. 
2) The section in the image shared Accept Incoming Change | Accept Current Change (Highlighted by yellow) is an option that allows which part of code we could retain. However, that clickable text becomes unclickable and you cannot even delete it.

**Comment by Singh-Krishna7** [comment]
The only option that I can think of is to downgrade or reinstall the previous version of VSCode to unblock myself from these issues at the moment.

**Comment by lorand-horvath** [comment]
Sorry for bringing this up again, but Git conflicts are not the topic of this issue. As far as I can see the OP is about experiencing lag when running concurrently.
Please open a different issue for git conflicts, otherwise I don't think any of them will be solved, if the reports are mixed together.

**Comment by bpasero** [comment]
Can people test this advice: https://github.com/microsoft/vscode/issues/146737#issuecomment-1090034221
